components/DocumentEditor.vue

```diff
@@ -100,53 +100,9 @@
 
 <script lang="ts">
 import { defineComponent, computed } from '@vue/composition-api'
-import { useResult, useQuery } from '@vue/apollo-composable'
+import { useResult } from '@vue/apollo-composable'
 import Tags from './tagify.vue'
-import { gql } from '~/apollo'
-
-export const DocumentDetails = gql(/* GraphQL */ `
-  fragment DocumentDetails on Document {
-    id
-    title
-    keywords
-    abstract
-    doi
-    authors {
-      ... on Person {
-        id
-        name
-      }
-      ... on Organization {
-        id
-        name
-      }
-    }
-    ... on JournalArticle {
-      in {
-        volume
-        number
-        journal {
-          id
-          name
-        }
-      }
-      published
-      pageStart
-      pageEnd
-    }
-    ... on ProceedingsArticle {
-      in {
-        title
-      }
-    }
-    ... on Thesis {
-      institution {
-        id
-        name
-      }
-    }
-  }
-`)
+import { useDocumentDetailsQuery } from '~/types/graphql'
```

Comment:
Since these are not pure types, I would prefer if the generated code is placed under `~/composables` (or if that leads to conflicts under `~/apollo`)